ownership/ownership.go

```diff
@@ -12,112 +12,114 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package keysync
+package ownership
 
 import (
-	"bufio"
 	"fmt"
-	"os"
+	"os/user"
 	"strconv"
-	"strings"
 
 	"github.com/sirupsen/logrus"
 )
 
+type Lookup interface {
+	UID(username string) (uint32, error)
+	GID(groupname string) (uint32, error)
+}
+
+type Os struct {}
+
+func (o Os) UID(username string) (uint32, error) {
+	u, err := user.Lookup(username)
+	if err != nil {
+		return 0, fmt.Errorf("Error resolving uid for %s: %v\n", username, err)
+	}
+	id, err := strconv.ParseUint(u.Uid, 10 /* base */, 32 /* bits */)
+	if err != nil {
+		return 0, fmt.Errorf("Error parsing uid %s for %s: %v\n", u.Uid, username, err)
+	}
+	return uint32(id), nil
+}
+
+func (o Os) GID(groupname string) (uint32, error) {
+	group, err := user.LookupGroup(groupname)
+	if err != nil {
+		return 0, fmt.Errorf("Error resolving gid for %s: %v\n", group, err)
+	}
+	id, err := strconv.ParseUint(group.Gid, 10 /* base */, 32 /* bits */)
+	if err != nil {
+		return 0, fmt.Errorf("Error parsing gid %s for %s: %v\n", group.Gid, groupname, err)
+	}
+	return uint32(id), nil
+}
+
+type Mock struct {
```

Comment:
Should I put the mock into a separate test package or something?not sure it's worth it for 15 lines of code.